assets/src/dashboard/app/views/myStories/content/storyGridItem/index.js

```diff
@@ -71,72 +67,124 @@ const StoryGridItem = ({
       }
     : {};
 
+  const isLocked = useMemo(() => {
+    return enablePostLocking && story?.locked && userId !== story?.lockUser.id;
+  }, [enablePostLocking, story, userId]);
+
   const generatedMenuItems = useMemo(
     () =>
       generateStoryMenu({
         menuItemActions: storyMenu.menuItemActions,
         menuItems: storyMenu.menuItems,
         story,
+        isLocked,
       }),
-    [storyMenu, story]
+    [storyMenu, story, isLocked]
   );
 
   const storyDate = getRelativeDisplayDate(
     story?.status === STORY_STATUS.DRAFT
       ? story?.modified_gmt
       : story?.created_gmt
   );
+
+  const formattedTitle = titleFormatted(story.title);
+
+  const memoizedStoryMenu = useMemo(
+    () => (
+      <StoryMenu
+        menuLabel={
+          isLocked
+            ? sprintf(
+                /* translators: %1$s: story title %2$s: user story is locked by.*/
+                __('Context menu for %1$s (locked by %2$s)', 'web-stories'),
+                formattedTitle,
+                story?.lockUser.name
+              )
+            : sprintf(
+                /* translators: %s: story title.*/
+                __('Context menu for %s', 'web-stories'),
+                formattedTitle
+              )
+        }
+        itemActive={isActive}
+        tabIndex={tabIndex}
+        onMoreButtonSelected={storyMenu.handleMenuToggle}
+        contextMenuId={storyMenu.contextMenuId}
+        storyId={story.id}
+        isInverted
+        menuItems={generatedMenuItems}
+        menuStyleOverrides={css`
```

Comment:
Elsewhere this same menu is used and it's positioned at the center not the bottom so just overriding.